### PR TITLE
Make fuel-core && fuel-gql-client optional deps

### PR DESCRIPTION
**Pull request by Voxelot**
The abi-gen-macro crate transitively includes fuel-core and fuel-gql-client even though those deps aren't used at all in the codegen process. This was causing circular dependencies when testing abi gen in the fuel-core indexer service. 

Ideally, codegen is refactored to live in fuels-core without requiring fuel-core or fuel-gql-client. I attempted this, but it was going to be too much of a major change, and resorted to feature flagging for now (basically the only thing left in fuels-rs after my refactor was the script module and some contract deployment and call functions).

### CODE REVIEW

**Review by rakita — DISMISSED** [review]
lgtm.
It seems okay that `fuels-rs` uses `fuel-core`,`fuel-gql-client`.
"Problem" (it is not even that problematic, it just add's one more step) is in example lib inside `fuel-core`. The solution is maybe to just remove it: https://github.com/FuelLabs/fuel-core/pull/114#discussion_r786328493

File: fuels-abigen-macro/Cargo.toml
Comment (adlerjohn) [inline fuels-abigen-macro/Cargo.toml:42]:
```suggestion
fuel-core = { version = "0.1", default-features = false }
fuel-gql-client = { version = "0.1", default-features = false }
```

**Review by adlerjohn — COMMENTED** [review]
CI is failing due to warnings being errors I think.

**Comment by Voxelot** [comment]
We can talk about where to move the example, but that will affect the infra team's progress so that should be coordinated with @tjsharp1. 

The abi gen macros are in a standalone crate used only for codegen of abi types, and these types have no integration or need to depend on fuel-core and fuel-gql-client. It's a lot of overhead to make everyone who wants to be able to codegen their abi types compile a lot of unrelated dependencies like graphql libraries etc.

**Review by adlerjohn — COMMENTED** [review]
lgtm, but will defer to @digorithm. Very ugly indeed 😂

**Review by tjsharp1 — APPROVED** [review]

**Review by digorithm — APPROVED** [review]

**Comment by Voxelot** [comment]
If it's too ugly I can close the PR and investigate a cleaner refactor. Moving the code-gen to fuels-core might be a pretty significant gutting of fuels-rs though.

**Comment by Voxelot** [comment]
Marking as a draft, for now, to allow @digorithm time to investigate other options and prevent someone from accidentally slapping the merge button.

**Comment by digorithm** [comment]
@Voxelot Do you mind closing this one since https://github.com/FuelLabs/fuels-rs/pull/66 was merged?

**Comment by Voxelot** [comment]
Gladly, thank you!

**Pull request closed without merging**
